typings.d.ts

```diff
@@ -0,0 +1,66 @@
+// WIP
+
+type PlacesType = 'city' | 'country' | 'address' | 'busStop' | 'trainStation' | 'townhall' | 'airport'
+
+type PlacesSuggestion = {
+  type: PlacesType,
+  name: string,
+  city: string,
```

Comment:
FIY, many fields in places can be optional, like `county`, `city`, `suburb`, `postcode`, `postcodes`, and `country`. I am pretty sure, I have even seen a few locations without any `administrative` data, as crazy as this may seem.

Some of these values can simply be missing because of the type of the suggestion.
I would maybe define 3 kind of suggestions depending on the type field.

```ts
type PlacesCountrySuggestion {
  type: 'country',
  name: string,
  countryCode: string,
  latlng: { lat: number, lng: number },
  value: string,
  highlight: {
    name?: string,
  }
}

type PlacesCitySuggestion {
  type: 'city',
  name: string,
  country: string,
  countryCode: string,
  administrative?: string,
  county?: string,
  latlng: { lat: number, lng: number },
  postcode?: string,
  postcodes?: string[],
  value: string,
  highlight: {
    name?: string,
    postcode?: string,
    county?: string,
    administrative?: string,
    country?: string
  }
}

type PlacesPointSuggestion {
  type: 'address' | 'busStop' | 'trainStation' | 'townhall' | 'airport' ,
  name: string,
  city?: string,
  country: string,
  countryCode: string,
  administrative?: string,
  county?: string,
  suburb?: string,
  latlng: { lat: number, lng: number },
  postcode?: string,
  postcodes?: string[],
  value: string,
  highlight: {
    name?: string,
    suburb?: string,
    city?: string,
    postcode?: string,
    county?: string,
    administrative?: string,
    country?: string
  }
}

type PlacesSuggestion = PlacesCountrySuggestion | PlacesCitySuggestion | PlacesPointSuggestion;
```

Reply:
Ok thanks Jonathan.

When you're saying optional or without any data, are you saying that the value will be empty (`''`), or it will not be present or equal to `undefined`?

I've quickly taken a look in the package and didn't see where the suggestion object was created.

Reply:
`undefined`

Reply:
Ok, thanks. It's clear with the code now (and actually it's not really `xxxx?: string` but `xxxx: string | undefined`. 
This is not a big thing, but can make a difference. (With `hasOwnProperty` for example)